### PR TITLE
Solution for Input meta ignored when creating a Table from a Table

### PR DESCRIPTION
**Pull request by himanshupathak21061998**
Fix: #8405
This PR will solve the issue by creating meta using both table meta as well as meta value passed as an argument.

### CODE REVIEW

**Comment by himanshupathak21061998** [comment]
Should I raise a ValueError in except of doing this because other tests are failing

File: astropy/table/table.py [hunk 1]
```diff
@@ -477,6 +477,12 @@ def __init__(self, data=None, masked=None, names=None, dtype=None,
                 # gets a key copy here if copy=False because later a direct object ref
                 # is used.
                 meta = data.meta if copy else data.meta.copy()
+            # If user-input meta is not None and data.meta also contains data
+            if meta is not None and data.meta:
+                if copy:
+                    meta = {**data.meta, **meta}
```
Comment (pllim) [inline astropy/table/table.py]: If your goal is to concat dictionaries, this is the wrong syntax.

Comment (himanshupathak21061998) [reply astropy/table/table.py]: Now using update() to concat

**Comment by himanshupathak21061998** [comment]
@pllim Failure is because of some test like this one
```
def test_table_meta_copy_with_meta_arg():
        """
        Test no copy vs light (key) copy vs deep copy of table meta when meta is
        supplied as a table init argument.  #8404.
        """
        meta = {1: [1, 2]}
        meta2 = {2: [3, 4]}
        t = table.Table([[1]], meta=meta, copy=False)
        assert t.meta is meta
    
        t = table.Table([[1]], meta=meta)  # default copy=True
        assert t.meta is not meta
        assert t.meta == meta
    
        # Test initializing from existing table with meta with copy=False
        t2 = table.Table(t, meta=meta2, copy=False)
       assert t2.meta is meta2
```
here meta2= {2: [3, 4]} but after changes done by me t2.meta = {2: [3, 4], 1: [1, 2] }
So, should I need to do here perform changes in previously written tests.

**Comment by pllim** [comment]
@himanshupathak21061998 , thanks. I think we should wait to hear from @taldcroft before proceeding.

**Comment by himanshupathak21061998** [comment]
@pllim I came up with this solution now we have to wait for @taldcroft for further process

**Comment by taldcroft** [comment]
Hi @himanshupathak21061998 - thanks for the PR!  Unfortunately my initial thought is that this is too small of a use case to justify a new top-level API update to the core `Table` class.  There is now an easy workaround.

The original #8405 should probably be closed since after #8404 it is possible to manually merge external `meta` and existing table meta.  Before #8404 it was impossible to do this because the table meta was always chosen.  E.g.
```
t = simple_table()
t.meta['a'] = 1
meta = {'b': 2}
meta.update(t.meta)  # This is what your `merge_meta` option would do
t2 = Table(t, meta=meta)
```

**Comment by himanshupathak21061998** [comment]
Closing it.

**Pull request closed without merging**